napari/utils/_proxies.py

```diff
@@ -41,21 +43,30 @@ class PublicOnlyProxy(wrapt.ObjectProxy, Generic[_T]):
     __wrapped__: _T
 
     def __getattr__(self, name: str):
-        if name.startswith('_'):
+        if _SUNDER.match(name):
+            # allow napari to access private attributes and get an non-proxy
+            frame = sys._getframe(1) if hasattr(sys, "_getframe") else None
+            if frame and misc.ROOT_DIR in frame.f_code.co_filename:  # type: ignore
```

Comment:
why `misc.ROOT_DIR in frame.f_code.co_filename` not `frame.f_code.co_filename.startswith(misc.ROOT_DIR)`?